### DISCUSSION TITLE
(Success!) Open Letter To Other Linux Users To Help Each Other Get Started Using GW2Radial

### DISCUSSION THREAD

**Discussion by jimmon89** [Friendly0Fire/GW2Radial#281] [Show and tell]
### Update
I haven't played GW2 in a while, so sorry for not noticing myself.
But as posted in the comments by the user @Zyian there are now official instructions to make GW2Radial work under wine/proton
[On the main page](https://github.com/Friendly0Fire/GW2Radial#linux-support)

### **Confirmed Working Environments:**
**Lutris - Link:**
[Main Page](https://lutris.net/games/guild-wars-2/)
[Installer Page](https://lutris.net/games/install/13197/view)
**Working Installer Name:** "DXVK 64-Bit version"
**Untested Version Name:** "Guild Wars 2 Armored Nvidia version"

**Proton Is Unconfirmed Whether It Works Or Not**
### **Confirmed Working Lutris Runners:**
lutris-fshack-6.21-6-x86_64
lutris-6.21-6-x86_64

### **Confirmed Not Working Lutris Runners:**
lutris-ge-6.21-1-x86_64

### **Confirmed Not Working Proton Versions:**
Proton-6.21-GE-2

### **Note:**
As More Users Post On This Thread With Their Info, If Any Of You Need More Information From Other Users I Ask That As Long As It Doesn't Break The Users Standard For Anonymity That You Please Share And I Will Do The Same.

**Comment by jimmon89** [comment]
I got it working!
I used Lutris and [this](https://lutris.net/games/install/13197/view) install script
afterwards, under "Configuration"
Under "game options"
Inside "arguments"
the default arguments are
`-dx11 -autologin`
which will make it default to DirectX 11
and must be changed to
`-dx9 -autologin`
for some reason just removing `-dx11` wasn't enough

after that GW2 Radial installed with "GW2 Addon Manager" worked flawlessly

**Comment by dhollinger** [comment]
How exactly did you get GW2 Addon Manager working in WINE? I just keep getting errors, even with .NET 4.x installed

**Comment by jimmon89** [comment]
> How exactly did you get GW2 Addon Manager working in WINE? I just keep getting errors, even with .NET 4.x installed

make sure you are using either "lutris-fshack-6.21-6-x86_64" or "lutris-6.21-6-x86_64"
if you used my recommended installer the default is "lutris-fshack-6.21-6-x86_64"
in Lutris on the side bar click the "Wine" runner
then select "Guild Wars 2"
then you should see a wine glass in the bottom bar
click the up arrow to the right of it and select "Run EXE inside Wine prefix"
then browse to where you unzipped the addon manager and run "GW2 Addon Manager.exe"
All of the ArcDPS addons install fine
GW2 Radial Installs Fine
libimgui installs fine
SelectRenderer installs fine
However,
BuildPad crashes the addon manager when i try to install it
And i have yet to try d912pxy so i can't speak for it and don't see a reason to install it
because then you'll be translating dx9 to dx12 to Vulcan and i can't imagine that would be any good for performance

###  **--Edit--**
after installing ArcDPS, all of it's addons, and SelectRenderer.
GW2 didn't seem to want to start
after i remove the SelectRenderer addon, plus the killproof and blish-hud ArcDPS addons it loaded fine
when i get a chance i'll see if it's just one or two of them that it didn't like or all

**Comment by dhollinger** [comment]
I get the Github API error over and over on the latest release.

**Comment by jimmon89** [comment]
that's not unique to linux
it's a known issue due to the current implementation of GW2 Addon Manager
see this issue [here](https://github.com/gw2-addon-loader/GW2-Addon-Manager/issues/150)
it's either that issue
or the github api key used by GW2 Addon Manager has used all of it's calls for now
if so it should fix it's self when api calls refresh

my money is on the API calls because i started getting that same error this afternoon

### **--edit--**
according to [this](https://github.com/gw2-addon-loader/GW2-Addon-Manager/issues/156) issue you may be able to get around api limits by temporarilly using a VPN

**Comment by dhollinger** [comment]
I got it figured out. I installed .NET Framework 4.7, then realized that I needed to disable DXVK to run the program. Now I don't have any issues.

For some reason .NET Framework 4.x doesn't like DXVK. Ran into similar issues with X-Wing Alliance Upgrade on lutris. Their installers, launcher, and tools don't play well with DXVK, but the mod requires DXVK to play the game

**Comment by jimmon89** [comment]
> I got it figured out. I installed .NET Framework 4.7, then realized that I needed to disable DXVK to run the program. Now I don't have any issues.
> 
> For some reason .NET Framework 4.x doesn't like DXVK. Ran into similar issues with X-Wing Alliance Upgrade on lutris. Their installers, launcher, and tools don't play well with DXVK, but the mod requires DXVK to play the game

very odd issue..
either way, i'm glad you got it working!

**Comment by Witte1985** [comment]
For those on Fedora

The new install script on lutris will not work if you have nvidia driver 495.46 combined with "lutris-fshack-6.21-6-x86_64" or "lutris-6.21-6-x86_64".
It was running fine on lutris-5.1 but the the AddonManager cannot be started on that.
using nvidia driver 470.94 combined with lutris-6.21 worked out just fine.

Just adding this bit of knowledge for whoever needs it.

However, regarding gw2radial. It is giving me an issue where it cannot  save the settings.
"Could not save addon configuration. Reason given was:"
"No such file or directory"

So i recursively set rwd rights for everyone  on the game folder, but no success. 
Where does gw2radial store its settings normally? probably an ini file or json file somewhere?

**Comment by jimmon89** [comment]
check the "Documents" folder for your prefix
for me it's at
`/media/ssd/games/guild-wars-2/drive_c/users/(snip username)/Documents/Guild Wars 2/addons/gw2radial`

**Comment by Witte1985** [comment]
@jimmon89 

Thanks for pointing me in the right direction.
there was however a second folder that needed elevated rights for me before the error dialogs disappeared.

~/Games/guild-wars-2/drive_c/users/(username)/Temp/gw2radial/gw2radial

Working perfectly now, thanks

**Comment by jimmon89** [comment]
Awesome! glad you got it working :)

**Comment by jimmon89** [comment]
I wanted to post an update 
I got GW2 Radial working with Proton under the version `Proton-7.1-GE-2` by adding the DLLs added by GW2 UOAOM to the override list
namely `dxgi=n,b;d3d9=n,b;d3d11=n,b;addonLoader.dll=n,b;d3dcompiler_47=n,b`
idk if addonLoader.dll=n,b is actually needed but, screw it, it works!

**Comment by dhollinger** [comment]
I've got this working in Linux with Steam Proton and Steam Guild Wars 2.

It does require you to install the `d3dcompiler_47` through protontricks and then set the following args in the launch options:

`WINEDLLOVERRIDES="dxgi=n,b;d3d9=n,b;d3d11=n,b;addonLoader.dll=n,b;d3dcompiler_47=n,b" %command%`

If you're logging in using a non-Steam account then you need to use this in the Launch Options:

`WINEDLLOVERRIDES="dxgi=n,b;d3d9=n,b;d3d11=n,b;addonLoader.dll=n,b;d3dcompiler_47=n,b" %command% -provider Portal`

If you are on Steam Deck, I would strongly encourage that you use Steam Input instead of GW2Radial. Steam Input can create Radial and Square menus natively using the Input API and touch pads. It will also save some work on your part

**Reply by jimmon89** [reply]
can confirm, i had the exact same experience
but i added the dlls into winecfg's library tab instead of using launch options

**Comment by Pakrohk** [comment]
I don't know if this problem is related to this topic or not.
I installed guild wars 2 on arch linux using steam and running it with ProtonGE 8.4.
I have no problem running the game, but when I install the gw2radial plugin, The game does not run, only the launcher logs in, and after clicking on the play button, I get the following error.

![image](https://github.com/Friendly0Fire/GW2Radial/assets/27810360/72fb8741-6e77-40b0-b6d2-109c5ad0c26b)

**Reply by Zyian** [reply]
https://github.com/Friendly0Fire/GW2Radial#linux-support
Use these instructions on how to fix this
